To: vim_dev@googlegroups.com
Subject: Patch 7.3.473
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.473
Problem:    'cursorbind' does not work correctly in combination with
	    'virtualedit' set to "all".
Solution:   Copy coladd. (Gary Johnson)
Files:	    src/move.c


*** ../vim-7.3.472/src/move.c	2012-02-04 23:34:57.000000000 +0100
--- src/move.c	2012-03-16 19:20:57.000000000 +0100
***************
*** 2843,2849 ****
  do_check_cursorbind()
  {
      linenr_T	line = curwin->w_cursor.lnum;
!     colnr_T	col =  curwin->w_cursor.col;
      win_T	*old_curwin = curwin;
      buf_T	*old_curbuf = curbuf;
      int		restart_edit_save;
--- 2843,2852 ----
  do_check_cursorbind()
  {
      linenr_T	line = curwin->w_cursor.lnum;
!     colnr_T	col = curwin->w_cursor.col;
! # ifdef FEAT_VIRTUALEDIT
!     colnr_T	coladd = curwin->w_cursor.coladd;
! # endif
      win_T	*old_curwin = curwin;
      buf_T	*old_curbuf = curbuf;
      int		restart_edit_save;
***************
*** 2875,2880 ****
--- 2878,2886 ----
  # endif
  		curwin->w_cursor.lnum = line;
  	    curwin->w_cursor.col = col;
+ # ifdef FEAT_VIRTUALEDIT
+ 	    curwin->w_cursor.coladd = coladd;
+ # endif
  
  	    /* Make sure the cursor is in a valid position.  Temporarily set
  	     * "restart_edit" to allow the cursor to be beyond the EOL. */
*** ../vim-7.3.472/src/version.c	2012-03-16 19:07:54.000000000 +0100
--- src/version.c	2012-03-16 19:24:06.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     473,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
201. When somebody asks you where you are, you tell them in which chat room.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
